src/components/application/life_cycle_charts/life_cycle_charts.jsx

```diff
@@ -119,65 +147,70 @@ class LifeCycleCharts extends React.Component {
-                    />
-                    <div style = {{flex: "auto"}}>
+                <div style = {{display: "flex", justifyContent: "flex-center"}}>
+                    <div style = {SIDE_STYLE}>
+                        <DateSelector
+                            datePicked   = {this.handleStartDatePicked}
+                            title        = 'Select Start Date'
+                            defaultValue = {this.state.totalViewPortDateRange.startDate}
+                            maxDate      = {this.state.totalViewPortDateRange.endDate}
+                        />
+                    </div>
+                    <div style = {MIDDLE_STYLE}>
                         <GlobalBrush
-                            show = {true}
-                            onBrushChange = {this.handleBrushChanged}
+                            show              = {true}
+                            onBrushChange     = {this.handleBrushChanged}
                             viewPortDateRange = {this.state.totalViewPortDateRange}
-                            height = {BRUSH_HEIGHT}
+                            height            = {BRUSH_HEIGHT}
                         />
                     </div>
-                    <div style={{width: '100px'}}>
-                        <DateSelector2
-                            datePicked = {this.handleEndDatePicked}
-                            title = 'Select Start Date'
-                            align = 'right'
+                    <div style = {SIDE_STYLE}>
+                        <DateSelector
+                            datePicked   = {this.handleEndDatePicked}
+                            title        = 'Select Start Date'
+                            align        = 'right'
                             defaultValue = {this.state.totalViewPortDateRange.endDate}
-                            minDate = {this.state.totalViewPortDateRange.startDate}
-                            maxDate = {moment().endOf('day').toDate()}
+                            minDate      = {this.state.totalViewPortDateRange.startDate}
+                            maxDate      = {moment().endOf('day').toDate()}
                         />
                     </div>
                 </div>
                 <hr />
-                <div style={{display: "flex", justifyContent: "flex-center"}}>
-                    <div style = {{flex: "auto"}}>
-                        <Charts
-                            selectedVersions = {this.state.selectedVerions}
-                            onDelete = {this.handleVersionRemoved}
-                            dataSets = {this.props.versionsHistory}
-                            viewPortDateRange = {this.state.zoomedViewPortDateRange}
-                            applicationId = {this.state.applicationId}
-                            height = {CHART_HEIGHT}
-                        />
-                    </div>
-                </div>
+                <Charts
```

Comment:
What is the `PropsExposer` for, why can't you use `react-dimensions` directly? They have some docs on how to use it with flexbox.